sycl/test-e2e/Annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,173 @@
+    auto ap = f(args);                                                         \
+    assert(ap.get() == nullptr);                                               \
+    free(ap, q);                                                               \
+  }
+
+constexpr int N = 10;
+
+template <typename T> void testUsmKind(sycl::queue &q) {
+  const sycl::context &Ctx = q.get_context();
+  auto dev = q.get_device();
+
+  TEST_USM_KIND(malloc_annotated<T>, alloc::device, N, q, alloc::device);
+  TEST_USM_KIND(malloc_annotated, alloc::device, N, q, alloc::device);
+  TEST_USM_KIND(aligned_alloc_annotated<T>, alloc::host, 1, N, q, alloc::host);
+  TEST_USM_KIND(aligned_alloc_annotated, alloc::host, 1, N, q, alloc::host);
+
+  TEST_USM_KIND(malloc_device_annotated, alloc::device, N, q);
+  TEST_USM_KIND(malloc_device_annotated<T>, alloc::device, N, q);
+  TEST_USM_KIND(aligned_alloc_device_annotated, alloc::device, 0, N, q);
+  TEST_USM_KIND(aligned_alloc_device_annotated<T>, alloc::device, 0, N, q);
+
+  TEST_USM_KIND(malloc_host_annotated, alloc::host, N, q);
+  TEST_USM_KIND((malloc_host_annotated<T>), alloc::host, N, q);
+  TEST_USM_KIND(aligned_alloc_host_annotated, alloc::host, 0, N, q);
+  TEST_USM_KIND((aligned_alloc_host_annotated<T>), alloc::host, 0, N, q);
+
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_USM_KIND(malloc_annotated, alloc::shared, N, q,
+                  properties{usm_kind_shared});
+    TEST_USM_KIND(malloc_annotated<T>, alloc::shared, N, q,
+                  properties{usm_kind_shared});
+
+    TEST_USM_KIND(malloc_shared_annotated, alloc::shared, N, q);
+    TEST_USM_KIND((malloc_shared_annotated<T>), alloc::shared, N, q);
+    TEST_USM_KIND(aligned_alloc_shared_annotated, alloc::shared, 0, N, q);
+    TEST_USM_KIND((aligned_alloc_shared_annotated<T>), alloc::shared, 0, N, q);
+  }
+}
+
+template <typename T> void testAlign(sycl::queue &q, unsigned align) {
+  const sycl::context &Ctx = q.get_context();
+  auto dev = q.get_device();
+
+  assert(align > 0 || (align & (align - 1)) == 0);
+
+  // Case: malloc_xxx with no alignment property, the alignment is sizeof(T)
+  TEST_ALIGN((malloc_device_annotated<T>), sizeof(T), N, q);
+  TEST_ALIGN((malloc_host_annotated<T>), sizeof(T), N, q);
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_ALIGN((malloc_shared_annotated<T>), sizeof(T), N, q);
+  }
+
+  // Case: malloc_xxx with compile-time alignment<0>, the alignment is sizeof(T)
+  TEST_ALIGN((malloc_device_annotated<T>), sizeof(T), N, q,
+             properties{alignment<0>});
+  TEST_ALIGN((malloc_host_annotated<T>), sizeof(T), N, q,
+             properties{alignment<0>});
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    TEST_ALIGN((malloc_shared_annotated<T>), sizeof(T), N, q,
+               properties{alignment<0>});
+  }
+
+  // Case: malloc_xxx with compile-time alignment<N> (N is not a power of 2),
```

Comment:
The non-power-of-2 case should also include tests for the Parameterized allocation functions, and the allocation functions requiring a usm_kind property.

Reply:
case added